mpb/bott.c

```diff
@@ -97,18 +97,26 @@ number_list bott_indices(integer band_start, integer n_bands)
     for (n = 1; n <= n_bands; ++n) { /* compute n-th bott index */
         sqmatrix_copy(Un, U);
         sqmatrix_copy(Wn, W);
+        Un.p = n_bands; /* sqmatrix_copy does not change the .p field (but _we_ do at every iteration */
+        Wn.p = n_bands; /* when we resize below), so to correct for that we must do it ourselves here */
```

Comment:
I don't understand, `sqmatrix_resize` already sets the `.p` field.  So why is this needed?

Reply:
When the matrices `Un` and `Wn` are copied at lines 98-99 their size still equals the `n` from the last iteration; then, when we do `sqmatrix_resize(..)` with `preserve_data = 1` the resize/copy loop is inconsistent because it refers to an erroneous size of the original matrix (`U` or `W`). 

I don't understand why the `sqmatrix_copy(..)` call doesn't fail though when `Un.p` and `U.p` (or, equivalently, `Wn` and `W`) must differ after the first iteration (i.e. why does the call to the below not fail at the CHECK?)
```c
/* A = B */
void sqmatrix_copy(sqmatrix A, sqmatrix B)
{
     CHECK(A.p == B.p, "arrays not conformant");
     blasglue_copy(A.p * A.p, B.data, 1, A.data, 1);
}
```

Reply:
Or, put more briefly: `sqmatrix_resize` only sets the `.p` field _after_ it has carried out the `preserve_data = 1` part: that part assumes that the `.p` field reflects the actual size of the original `.data` field.

Reply:
The CHECK issue seems to have originated with this commit https://github.com/stevengj/mpb/commit/225be5678edf65ba144f5bcf3aa341383fb2cb94#diff-c1263b161d86930f3050f951a96fe060 (line 63). I had been comparing to the `sqmatrix_copy(..)` on the master branch, rather than the bott branch; silly. I don't recall why we made that change then.

Anyway, I've reverted that change and reinstated the original CHECK and moved the `n_bands`-reassignment of the `.p` fields up so it's made before we call `sqmatrix_copy(..)`. This seems to provide consistent fix?